scripts/geninstr.ts

```diff
@@ -289,7 +302,7 @@ function generateTypes(genInfo: GenInfo, doc?: string) {
     }
     dtsProps.push(`  ${jsdoc}${genInfo.js}${q}: ${returnType};`);
   } else {
-    const paramStrs = params.map((v) => `${v.name}: ${v.type}`);
+    const paramStrs = params.map((v) => `${v.name}${v.optional ? '?' : ''}: ${v.type}`);
```

Comment:
Not really related to blueprint functionality, but the `?` appended to the name was also appearing in the JSDoc which is invalid. This keeps the original name in the JSDoc.